libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2339,6 +2339,202 @@ bool WaveTrack::Get(size_t iChannel, size_t nBuffers,
    });
 }
 
+namespace {
+void RoundToNearestClipSample(const WaveTrack& track, double& t)
+{
+   const auto clip = track.GetClipAtTime(t);
+   if (!clip)
+      return;
+   t = clip->SamplesToTime(clip->TimeToSamples(t - clip->GetPlayStartTime())) +
+       clip->GetPlayStartTime();
```

Comment:
`WaveClip::SnapToSample` analogous to the `WideSampleSequence` member function might be nice

Reply:
It might easily be confused for `WaveTrack::SnapToSample`, though. This manipulation is safe here in this `RoundToNearestClipSample` function, I'd leave it as it is if there aren't other needs elsewhere for it.